source/mbed_trace.c

```diff
@@ -454,8 +454,12 @@ void mbed_vtracef(uint8_t dlevel, const char* grp, const char *fmt, va_list ap)
 
 end:
     if ( m_trace.mutex_release_f ) {
-        while (m_trace.mutex_lock_count > 0) {
-            m_trace.mutex_lock_count--;
+        // Store the mutex lock count to temp variable so that it won't get
```

Comment:
"temp_mutex_lock_count" is long-winded to the point of making it hard to read - looks visually like the global.

Just "count" would do.

Could make it a "do/while" loop, as we "know" count is at least 1, and we hold the mutex at least once.

Logic seems solid though. (But then I didn't spot the problem in the last one either.)